server/app/repositories/workspace.py

```diff
@@ -95,3 +95,21 @@ async def get_space_datasets(self, workspace_id: str) -> List[Dataset]:
             .filter(DatasetSpace.workspace_id == workspace_id)
         )
         return result.unique().scalars().all()
+
+    async def get_users_by_workspace_id(self, workspace_id: str):
+        users = await self.session.execute(
+            select(UserSpace)
+            .options(joinedload(UserSpace.user))
+            .filter(UserSpace.workspace_id == workspace_id)
+        )
+        user_spaces = users.scalars().all()
+        result = []
+        for user_space in user_spaces:
```

Comment:
@shahrukh802 you don't need this loop for filtering you can handle that with pydantic to select particular fields check app/schema/responses

Reply:
resolved